dbt/include/spark/macros/adapters.sql

```diff
@@ -175,12 +175,20 @@
   {% do return(load_result('get_columns_in_relation').table) %}
 {% endmacro %}
 
-{% macro spark__list_relations_without_caching(relation) %}
-  {% call statement('list_relations_without_caching', fetch_result=True) -%}
-    show table extended in {{ relation }} like '*'
+{% macro spark__list_tables_without_caching(relation) %}
+  {% call statement('list_tables_without_caching', fetch_result=True) -%}
+    show tables in {{ relation.schema }}
   {% endcall %}
 
-  {% do return(load_result('list_relations_without_caching').table) %}
+  {% do return(load_result('list_tables_without_caching').table) %}
+{% endmacro %}
+
+{% macro spark__list_views_without_caching(relation) %}
+  {% call statement('list_views_without_caching', fetch_result=True) -%}
+    show views in {{ relation.schema }}
```

Comment:
Sounds like `show views` is officially new in Spark v3: https://issues.apache.org/jira/browse/SPARK-31113

So we need to:
- Make a call on whether Spark v3 is the minimum required for use with new versions of `dbt-spark`. I think that's reasonable, it's been ~2 years since its final release.
- Upgrade the version of Spark that's running in `integration-spark-thrift`. This is something we've struggled to do before. If anyone could help out here, I'd really appreciate it!

Reply:
I did some research and found this to be a strange problem. Now `integration-spark-session` doesn't pass, I see that it's running in `godatadriven/pyspark:3.1`, I go into the image and see that the Spark version is v3.1.2, then I print `views.column_names` and get: `<agate.MappedSequence: ()>`. As described in the Jira URL,  `show views in` should have been supported in Spark v3.0.0, I have no idea about it, I'll try to be compatible with this situation first.

Also for `integration-spark-thrift` I will also upgrade the Spark image it depends on.

Reply:
The interesting thing happened, I went into the image and spark-submit a python file of show views directly, which returned the normal result.

python file:
```
from pyspark.sql import SparkSession

spark = SparkSession.builder.getOrCreate()

spark.sql('show views in default').show()
```

result:
```
+---------+--------+-----------+
|namespace|viewName|isTemporary|
+---------+--------+-----------+
+---------+--------+-----------+
```

Reply:
@JCZuurmond Could I ask for your help here, in understanding the behavior of `show views` in the `session` connection method? My sense is that [`SessionConnectionWrapper._cursor.execute`](https://github.com/dbt-labs/dbt-spark/blob/f03d217ebb004bb889eff80d61da14bc72983370/dbt/adapters/spark/session.py#L87-L110) should just be submitting the SQL passed in.

@TalkWIthKeyboard For context on what's going on here, `execute_macro` is just an indirect way of calling `execute`:
- `execute_macro` does what it says: it executes the macro, via `MacroGenerator`, and returns the result ([source code in `dbt-core`](https://github.com/dbt-labs/dbt-core/blob/788694ec5b9796d45592a8e704a3618881775e2a/core/dbt/adapters/base/impl.py#L930-L934))
- these macros (`spark__list_tables_without_caching` + `spark__list_views_without_caching`), in turn, call `statement`s and fetch the results
- a `statement` really just passes back to `adapter.execute` ([source code in `dbt-core`](https://github.com/dbt-labs/dbt-core/blob/788694ec5b9796d45592a8e704a3618881775e2a/core/dbt/include/global_project/macros/etc/statement.sql#L10))

Reply:
Fixed it, there is no problem with `execute_macro`. I used `agate.Table` incorrectly, leading to my misunderstanding of the error, anyway, thanks for your help.

Reply:
I have fixed the issue, it had nothing to do with `execute_macro`, I was not familiar with `agate.Table` so I used it incorrectly which misled me about the issue. Anyway, thank you very much for your help.

Reply:
Do you still want my input? Looks like you resolved it

Reply:
> Do you still want my input? Looks like you resolved it

No, thanks, I resolved it.